### PR TITLE
Debounce waiting for supermajority to try to prevent nodes from starting too early

### PR DESCRIPTION
**Pull request by mvines**
Not all nodes will observe 80% at the same time. 
I came up with a 15 second delay here, seemed reasonable but maybe too short?

### CODE REVIEW

**Comment by t-nelson** [comment]
doesn't seem unreasonable.  @behzadnouri 15s sufficient for contactinfo to circulate?

**Comment by mvines** [comment]
hmmm, I'm now wondering if this would even help at all.

If a node reaches 80% before the rest and resumes early, it'll remain in gossip for the rest to see eventually anyway. Adding a simple delay at 80% to wait for the rest just pushes everybody back by that delay.

Maybe a better approach (over just closing this PR) would be to delay at least 15 seconds and then round up to the next UTC minute (or 30 second mark).

**Comment by t-nelson** [comment]
seems like that could still cut us pretty bad if observation is right around a minute boundary.  

maybe need a way to just go if we see blocks produced and are withing N%

File: core/src/validator.rs [hunk 1]
```diff
@@ -1912,11 +1914,32 @@ fn wait_for_supermajority(
...
+                            "{}. Waiting {} seconds for gossip to stabilize",
+                            msg_prefix,
+                            SUPERMAJORITY_OBSERVATION_THRESHOLD_DURATION.as_secs()
+                        );
+                    }
+                } else {
+                    if supermajority_first_observed.is_some() {
+                        info!("Supermajority lost");
+                    }
+                    supermajority_first_observed = None;
```
Comment (mvines) [inline core/src/validator.rs:1942]: Probably need to debounce this case as well

**Comment by behzadnouri** [comment]
> doesn't seem unreasonable. @behzadnouri 15s sufficient for contactinfo to circulate?

I believe so. coincidentally that is the same as `CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS`:
https://github.com/search?q=repo%3Asolana-labs%2Fsolana%20CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS&type=code

**Comment by mvines** [comment]
I'm thinking I'll cool my jets on this one, I'm no longer convinced this is an issue worth messing around in the restart code for. Thanks for checking it out

**Comment by steviez** [comment]
While this is still fresh, some extra comment for the sake of our future selves.

Suppose a cluster is at `(80 - x)%` active stake in gossip waiting for supermajority, and some node `N` joins with `> x%` stake. `N` will push the cluster over the 80% threshold; additionally, `N` will have a headstart on normal operation as the rest of the cluster will have to wait for `N` to become visible in gossip.

Our concern was something along the lines of `N` proceeding through wait-for-supermajority too quickly for other nodes to see it (potentially forking off or something). Despite the headstart, `N` will become visible to the rest of the cluster in the near future and be counted by below function, regardless of whether `N` is in the wait-for-supermajority state or past it already:
https://github.com/solana-labs/solana/blob/cd93719f68a20e78bff9f0104860cdf6d914f049/core/src/validator.rs#L1934

So, the "headstart" that `N` might get does not seem to be problematic like we initially thought [in Discord](https://discord.com/channels/428295358100013066/670512312339398668/1043210414818865173).

**Pull request closed without merging**